### ISSUE TITLE
Parity between HTTP / PB requests and responses [JIRA: RIAK-1531]

### ISSUE THREAD

**Issue by Basho-JIRA** [basho/yokozuna#451]
Currently, HTTP requests and responses for Yokozuna contain richer information than PB. This is problematic if we are encouraging use of PB in clients, meaning that they get reduced functionality.

Possible solutions include:
- Re-encoding Solr response with more free-form messages
- Encoding the PB response on the Solr side and passing through Riak PB interface to client

**[Created in JIRA by Sean Cribbs]**

**Comment by broach** [comment]
The only thing needed to fix this is a type enum in the `RpbPair` protobuf for the value (and of course, filling out said field in Riak/YZ). 

```
message RpbPair {
    enum ValueType {
        null = 0;
        boolean = 1;
        int = 2;
        float = 3;
        string = 4;
    }
    required bytes key = 1;
    optional bytes value = 2;
    optional ValueType vtype = 3; 
}
```

This would allow backward compatibility and fix the glitch.

**Comment by broach** [comment]
Yes, the null bit is somewhat kludgey but it's for backward compatibly -  today we're sending a string `"null"`
